dotcom-rendering/src/server/prod-server.ts

```diff
@@ -52,79 +52,43 @@ export const prodServer = (): void => {
 		app.use('/assets', express.static(__dirname));
 	}
 
-	app.post('/Article', logRenderTime, renderArticle);
-	app.post('/AMPArticle', logRenderTime, renderAMPArticle);
-	app.post('/Interactive', logRenderTime, renderInteractive);
-	app.post('/AMPInteractive', logRenderTime, renderAMPArticle);
-	app.post('/Blocks', logRenderTime, renderBlocks);
-	app.post('/KeyEvents', logRenderTime, renderKeyEvents);
-	app.post('/Front', logRenderTime, renderFront);
-	app.post('/FrontJSON', logRenderTime, renderFrontJson);
+	app.post('/Article', logRenderTime, handleArticle);
+	app.post('/AMPArticle', logRenderTime, handleAMPArticle);
+	app.post('/Interactive', logRenderTime, handleInteractive);
+	app.post('/AMPInteractive', logRenderTime, handleAMPArticle);
+	app.post('/Blocks', logRenderTime, handleBlocks);
+	app.post('/KeyEvents', logRenderTime, handleKeyEvents);
+	app.post('/Front', logRenderTime, handleFront);
+	app.post('/FrontJSON', logRenderTime, handleFrontJson);
 
 	// These GET's are for checking any given URL directly from PROD
 	app.get(
 		'/Article',
 		logRenderTime,
 		getContentFromURLMiddleware,
-		async (req: Request, res: Response) => {
-			// Eg. http://localhost:9000/Article?url=https://www.theguardian.com/commentisfree/...
-			try {
-				return renderArticle(req, res);
-			} catch (error) {
-				console.error(error);
-			}
-		},
+		handleArticle,
 	);
+	app.use('/ArticleJson', handleArticleJson);
 
 	app.get(
 		'/AMPArticle',
 		logRenderTime,
 		getContentFromURLMiddleware,
-		async (req: Request, res: Response) => {
-			// Eg. http://localhost:9000/AMPArticle?url=https://www.theguardian.com/commentisfree/...
-			try {
-				return renderAMPArticle(req, res);
-			} catch (error) {
-				console.error(error);
-			}
-		},
-	);
-
-	app.get(
-		'/Front',
-		logRenderTime,
-		// TODO: ensure getContentFromURLMiddleware supports fronts
-		getContentFromURLMiddleware,
-		async (req: Request, res: Response) => {
-			// Eg. http://localhost:9000/Front?url=https://www.theguardian.com/uk/sport
-			try {
-				return renderFront(req, res);
-			} catch (error) {
-				console.error(error);
-			}
-		},
+		handleAMPArticle,
 	);
 
+	app.get('/Front', logRenderTime, getContentFromURLMiddleware, handleFront);
 	app.get(
 		'/FrontJSON',
 		logRenderTime,
-		// TODO: ensure getContentFromURLMiddleware supports fronts
 		getContentFromURLMiddleware,
-		async (req: Request, res: Response) => {
-			// Eg. http://localhost:9000/FrontJSON?url=https://www.theguardian.com/uk/sport
-			try {
-				return renderFrontJson(req, res);
-			} catch (error) {
-				console.error(error);
-			}
-		},
+		handleFrontJson,
 	);
 
-	app.use('/ArticlePerfTest', renderArticlePerfTest);
-	app.use('/AMPArticlePerfTest', renderAMPArticlePerfTest);
-	app.use('/ArticleJson', renderArticleJson);
+	app.use('/ArticlePerfTest', handleArticlePerfTest);
+	app.use('/AMPArticlePerfTest', handleAMPArticlePerfTest);
 
-	app.get('/', (req: Request, res: Response) => {
+	app.get('/', (_, res) => {
```

Comment:
Do we want to add types here? Looking at the `handleError` method below where we have the `Reponse` type.

Reply:
No, the types can be inferred from the fact that his has to be a express handler!

I’ve now removed the type from the `handleError` method, too.

Reply:
Fair enough, even if the parameter is ignored, it keeps the signature more consistent.